src/cs50/sql.py

```diff
@@ -125,6 +131,12 @@ def execute(self, sql, *args, **kwargs):
             if token.ttype in [sqlparse.tokens.Keyword.DDL, sqlparse.tokens.Keyword.DML]:
                 command = token.value.upper()
                 break
+
+            # Begin a new session, if transaction started by caller (not using autocommit)
+            elif token.value.upper() in ["BEGIN", "START"]:
+                if self._session is not None:
+                    self._session.close()
+                self._session = self._Session()
```

Comment:
Would that fail quietly if the user tries to execute > 1 `BEGIN|START` statements without a `COMMIT|ROLLBACK` in between? Should it?

Reply:
It is a quiet failure -- not sure if it necessarily should be. I guess it's a balance between being kind to beginners (as by closing and rolling back, i.e. current implementation) and teaching beginners (as by throwing an error saying they didn't close the previous transaction). The latter is probably better, so I can change this to throw an exception if preferred.

Reply:
I'm leaning toward raising an exception but I'll defer to @dmalan as to what he thinks is better pedagogically.

Reply:
I'd raise a `RuntimeError("transaction already open")`? Seems fine for library to support only single transactions.

Reply:
Added to local copy